### PR TITLE
Moved picking quiz code to utils and supportive component (#134) (#177)

### PR DESCRIPTION
**Pull request by Derstilon**
Just a minor refactorization. I've moved some lines of code to the two new files.
image

### CODE REVIEW

File: src/lecturer/components/pickQuizView/PickQuizList.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+import { Search } from "@material-ui/icons";
+interface QuizListViewProps {
+    title: React.ReactNode;
+    items: number[];
+    isQuiz: boolean;
+    numberOfChecked: (items: number[]) => number;
+    checked: number[];
+    questions: Question[];
+    handles: any;
```
Comment (ostatni5) [inline src/lecturer/components/pickQuizView/PickQuizList.tsx:24]: probably should be an array of functions type or better an object

File: src/lecturer/components/pickQuizView/PickQuizList.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    const title = props.title;
+    const items = props.items;
+    const isQuiz = props.isQuiz;
+    const numberOfChecked = props.numberOfChecked;
+    const [handleToggleAll, handleChange, handleToggle, handleSearch] = [
+        ...props.handles,
+    ];
+    // const handleChange = props.handleChange;
+    // const handleToggle = props.handleToggle;
+    // const handleSearch = props.handleSearch;
```
Comment (ostatni5) [inline src/lecturer/components/pickQuizView/PickQuizList.tsx:38]: comments to remove?

File: src/lecturer/components/pickQuizView/PickQuizView.tsx [hunk 3]
```diff
@@ -318,7 +309,23 @@ export function PickQuizView() {
...
+                        handles={[
+                            handleToggleAll,
+                            handleChange,
+                            handleToggle,
+                            handleSearch,
+                        ]}
+                        checked={checked}
+                        questions={questions}
+                    />
+                    {/* {customList("Lista pytań", filterFn.fn(left), false)} */}
```
Comment (ostatni5) [inline src/lecturer/components/pickQuizView/PickQuizView.tsx:327]: comment?

File: src/lecturer/components/pickQuizView/PickQuizView.tsx [hunk 4]
```diff
@@ -343,7 +350,23 @@ export function PickQuizView() {
...
+                        handles={[
+                            handleToggleAll,
+                            handleChange,
+                            handleToggle,
+                            handleSearch,
+                        ]}
+                        checked={checked}
+                        questions={questions}
+                    />
+                    {/* {customList("Quiz", right, true)} */}
```
Comment (ostatni5) [inline src/lecturer/components/pickQuizView/PickQuizView.tsx:368]: comment?

**Review by ostatni5 — APPROVED** [review]
Fix problems and you are good to go 
obraz

File: src/lecturer/components/pickQuizView/PickQuizList.tsx (hunk 1 above)
Comment (Derstilon) [reply src/lecturer/components/pickQuizView/PickQuizList.tsx:24]: image
Is this better?

File: src/lecturer/util/boolAlgebra.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    return a.filter((value) => b.indexOf(value) === -1);
+}
+
+export function intersection(a: number[], b: number[]) {
+    return a.filter((value) => b.indexOf(value) !== -1);
+}
+
+export function union(a: number[], b: number[]) {
+    return [...a, ...not(b, a)];
+}
```
Comment (michal367) [inline src/lecturer/util/boolAlgebra.ts:11]: I like this file

**Review by michal367 — APPROVED** [review]
obraz

**Pull request merged**